ring/partition_ring_watcher.go

```diff
@@ -106,6 +107,15 @@ func (w *PartitionRingWatcher) updatePartitionRing(desc *PartitionRingDesc) {
 	for state, count := range desc.countPartitionsByState() {
 		w.numPartitionsGaugeVec.WithLabelValues(state.CleanName()).Set(float64(count))
 	}
+
+	// Check partitions whose state change is locked and log them.
+	for partitionID, partition := range desc.Partitions {
+		state := partition.GetState().CleanName()
+		partitionIDStr := strconv.Itoa(int(partitionID))
+		if partition.StateChangeLocked {
+			level.Warn(w.logger).Log("msg", "partition state change is locked", "partition_id", partitionIDStr, "partition_state", state)
```

Comment:
This is extremely verbose. Why not just logging when it has changed compared to the `oldRing`?